### PR TITLE
feat: remove overrideMethod middleware

### PR DESCRIPTION
**Pull request by fengmk2**
It's not safe it enable by default in egg

##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: app/middleware/site_file.js [hunk 1]
```diff
@@ -6,10 +6,10 @@ const MAX_AGE = 'public, max-age=2592000'; // 30 days
 module.exports = options => {
   return function* siteFile(next) {
     if (this.method !== 'HEAD' && this.method !== 'GET') return yield next;
-
-    if (!options.hasOwnProperty(this.path)) return yield next;
+    if (this.path[0] !== '/') return yield next;
```
Comment (fengmk2) [inline app/middleware/site_file.js]: 加个字符串判断，就不担心 hasOwnProperty 被入注了

**Comment by popomore** [comment]
这个也是不兼容更新，内部要更新下。还有为啥去掉要写一下

**Comment by fengmk2** [comment]
@popomore 在内部 egg issue 回复了

**Comment by fengmk2** [comment]
@popomore 这个没问题就合并发个版本。

**Comment by popomore** [comment]
+ egg@0.11.0

**Pull request merged**